confirm.c

```diff
@@ -763,9 +855,35 @@ confirm_query_create4(const struct sockaddr_storage *dst, uint8_t ttl,
 }
 
 struct confirm_query *
+confirm_query_create6_tcp(const struct sockaddr_storage *dst, uint8_t ttl,
+		uint8_t traffic_class, uint32_t flow_label, uint8_t flowid,
+		uint16_t src_port, uint16_t dst_port, confirm_query_cb cb)
+{
+	struct confirm_query *query = confirm_query_create6(dst, ttl, traffic_class,
+			flow_label, flowid, cb);
+
+	query->src_port = src_port;
+	query->dst_port = dst_port;
+	query->type = QUERY_TYPE_TCP;
+	return query;
+}
+
+struct confirm_query *
+confirm_query_create6_icmp(const struct sockaddr_storage *dst, uint8_t ttl,
+		uint8_t traffic_class, uint32_t flow_label, uint16_t icmpid,
+		uint8_t flowid, confirm_query_cb cb)
```

Comment:
the previous version of netlib already worked that way. I also think we should have icmp_seq here, we should have seq_number in confirm_query_create6_tcp too. Since we are packing data in these fields, they dont appear here. I think we should remove data_pack and data_unpack from netlib, since these functions are not generic (they are specific to our application paris6). That way we can add these fields to confirm_query_create6_icmp and confirm_query_create6_tcp. Can I do that?